### PR TITLE
NAS-132996 / 25.04 / Add fibre channel connections card

### PR DESCRIPTION
**Pull request by denysbutenko**
**Changes:**

Add fibre channel connections card

**Testing:**

Use m50 for testing. Check Target list and `fc1` to see `Fibre Channel Connections` card.

### Downstream

|Affects         |Reasoning
|----------------|-------------------------------
|Documentation   | New component

### CODE REVIEW

**Comment by denysbutenko** [comment]
time 16:30

File: src/app/pages/sharing/iscsi/target/all-targets/target-details/fibre-channel-connections-card/fibre-channel-connections-card.component.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    TranslateModule,
+  ],
+})
+export class FibreChannelConnectionsCardComponent implements OnInit {
+  connections = input<FibreChannelStatus[]>([]);
+
+  constructor(private api: ApiService) { }
+
+  ngOnInit(): void {
+    console.info('init');
```
Comment (AlexKarpov98) [inline src/app/pages/sharing/iscsi/target/all-targets/target-details/fibre-channel-connections-card/fibre-channel-connections-card.component.ts:31]: please remove it

File: src/app/pages/sharing/iscsi/target/all-targets/target-details/target-details.component.ts [hunk 2]
```diff
@@ -32,6 +35,7 @@ export class TargetDetailsComponent {
   readonly target = input.required<IscsiTarget>();
 
   targetPort = signal<FibreChannelPort>(null);
+  connections = toSignal(this.api.call('fcport.status'));
```
Comment (AlexKarpov98) [inline src/app/pages/sharing/iscsi/target/all-targets/target-details/target-details.component.ts:38]: there probably may be same connections? Maybe we can insure that we only show unique connections?

I did it for iSCSI sessions:

```
.subscribe((sessions) => {
        // Remove duplicates using lodash's uniqBy
        const uniqueSessions = uniqBy(sessions, (session) => `${session.initiator}_${session.initiator_addr}`);
        this.sessions.set(uniqueSessions);
      });
```

Comment (AlexKarpov98) [inline src/app/pages/sharing/iscsi/target/all-targets/target-details/target-details.component.ts:38]: maybe we can put this call directly inside `ix-fibre-channel-connections-card`?

**Review by AlexKarpov98 — APPROVED** [review]
Card Works good.
Screenshot 2024-12-17 at 12 44 38

Please see my comments.

File: src/app/pages/sharing/iscsi/target/all-targets/target-details/target-details.component.ts (hunk 2 above)
Comment (denysbutenko) [reply src/app/pages/sharing/iscsi/target/all-targets/target-details/target-details.component.ts:38]: This is not a case. Here sessions transformed into boolean. Anyway thanks for the tip.

**Pull request merged**